### PR TITLE
Adding a CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by dcoeurjo**
This file describes how to contribute to the library (coding style, requirements,...).

Once added to DGtal, Github users will be warned that a guideline exists when creating a Pull-Request (see 
https://help.github.com/articles/setting-guidelines-for-repository-contributors/)

### CODE REVIEW

**Comment by dcoeurjo** [comment]
ping @JacquesOlivierLachaud @kerautret

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+master branch. Note that we use [Travis](http://travis-ci.org) as
+Continuous Integration platform. When creating a pull-request, several
+builds are tested on several systems (linux, macos), compilers (gcc,
+clang) and configurations (DGtal dependencies). Before being merged, a
+pull-request must pass all these tests.
+
+More information are described in the
+[DGtal documentation](http://dgtal.org/doc/stable/moduleFAQGit.html).
+
+## Code Strucutre
```
Comment (JacquesOlivierLachaud) [inline CONTRIBUTING.md]: structure

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  namespaces that contain predefined types for digital geometry in
+  dimension 2 and 3 (respectivelly) using an arithmetical kernel based
+  on ```DGtal::int32_t```.
+* ```DGtal::deprecated::``` contains deprecated classes  and functions.
+* ```DGtal::detail::``` contains some internal classes and funcitons.
+
+
+## Coding style
+
+We exepct the source code to match with some coding style rules
```
Comment (JacquesOlivierLachaud) [inline CONTRIBUTING.md]: expect

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      }
+      finalthing();
+    }
+
+    unsigned int myMember;
+};
+```
+### Naming rules
+
+* Types/methods/variable/classes in ```CamelCase```: (e.g.
```
Comment (JacquesOlivierLachaud) [inline CONTRIBUTING.md]: methods and variables start lowercase.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+### Other rules
+
+* All file must contain a proper LGPL/GPL header. See the source files for examples.
+* Class member accessors/mutators
+ * accessors are usually not prefixed by "get" is is name by the class member
+ * if the accessor needs some computations, it is prefixed by "get"
+
+      `Value parameter()` (if a `Value myParamter` exists)
+
+      `Value getParemeter()` (there is "computation")
```
Comment (JacquesOlivierLachaud) [inline CONTRIBUTING.md]: getParameter

**Comment by JacquesOlivierLachaud** [comment]
Sorry I just have the time to correct english/typos but not the formatting.

**Comment by dcoeurjo** [comment]
no problem. thanks for the feedbacks.
Do you see if something is missing?

File: CONTRIBUTING.md (hunk 3 above)
Comment (dcoeurjo) [inline CONTRIBUTING.md]: done.

**Comment by JacquesOlivierLachaud** [comment]
Well it's a good starting point, and we can enrich it later on. We also have other implicit coding rules which are rather natural (i,j,k,l,m,n are integers, x,y,z real numbers or points, and so on)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+with the documentation...).
+
+## Contribution reviewing process
+
+We process contributions to the library using GitHub Pull-request
+mechanism. If you want to contribute, you would have to:
+* create an account on GitHub;
+* clone [DGtal](https://github.com/DGtal-team/DGtal) main repository
+  (or [DGtalTools](https://github.com/DGtal-team/DGtalTools),
+  [DGtalTools-contrib](https://github.com/DGtal-team/DDGtalTools-contrib));
```
Comment (kerautret) [inline CONTRIBUTING.md]: a "D" to be removed:
https://github.com/DGtal-team/DGtalTools-contrib

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+* a new entry in the project ```Changelog.md``` file.
+
+Once the pull-request has been sent, a DGtal package manager will
+review the code and may ask for edits before being merged to the DGtal
+master branch. Note that we use [Travis](http://travis-ci.org) as
+Continuous Integration platform. When creating a pull-request, several
+builds are tested on several systems (linux, macos), compilers (gcc,
+clang) and configurations (DGtal dependencies). Before being merged, a
+pull-request must pass all these tests.
+
```
Comment (kerautret) [inline CONTRIBUTING.md]: perhaps also add windows and appveyor platform ?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+* ```cmake```contains all [cmake](http://cmake.org) configuration
+  scripts.
+
+### Namespaces
+
+All DGtal classes and functions are contained in a ```DGtal::``` C++
+namespace. Additionally, sub-namespaces are used to gather specific
+codes:
+* ```DGtal::functions::``` contains all DGtal global functions.
+* ```DGtal::functors::``` contains all C++ functors.
```
Comment (kerautret) [inline CONTRIBUTING.md]: -> all DGtal functors ?

**Comment by kerautret** [comment]
ok fine for just see some typo ;)

**Comment by dcoeurjo** [comment]
Thanks. Typos fixed. merging.

**Pull request merged**